File: src/main/java/io/github/cootshk/quicksearch/impl/Searchable.java
package io.github.cootshk.quicksearch.impl;

import io.github.cootshk.quicksearch.api.SearchableType;
import net.minecraft.network.chat.Component;

public interface Searchable {
    Component quickSearch$getName();
    String quickSearch$getIdentifier();
    SearchableType quickSearch$getSearchableType();

    default Component getName() {
        return quickSearch$getName();
    }
    default String getSearchString() {
        return getName().getString();
    }
    default String getIdentifier() {
        return quickSearch$getIdentifier();
    }
    default SearchableType getType() {
        return quickSearch$getSearchableType();
    }
}

File: src/main/kotlin/io/github/cootshk/quicksearch/util/Searchable.kt
package io.github.cootshk.quicksearch.util

import io.github.cootshk.quicksearch.ui.SearchResult
import io.wispforest.owo.ui.core.Component
import net.minecraft.item.Item
import net.minecraft.text.Text

data class Searchable(val name: Text, private val _description: Text?, val render: Component) {
    constructor(name: Text, render: Component) : this(name, null, render)
    val description get() = _description ?: Text.translatable("quicksearch.description.default") as Text
    override fun toString(): String {
        return description.toString()
    }

    constructor(item: Item) : this(
        item.name,
        Text.of(Text.translatable("quicksearch.description.default")),
        SearchResult.of(item)
    )
}
